teps/0091-trusted-resources.md

```diff
@@ -364,26 +372,137 @@ type KeylessRef struct {
+ ```yaml
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: feature-flag
   namespace: tekton-pipelines
 data:
-  verification-policy: enforce
+  trusted-resources-verification-no-match-policy: "fail"
 ```
 
-`verification-policy`. (Optional, `enforce`, `warn` or `skip`, default `skip`):
- * `enforce`: Failing verification will mark the taskruns/pipelineruns as failed.
- * `warn`: Log warning but don't fail the taskruns/pipelineruns.
- * `skip`: Directly skip the verification.
+`trusted-resources-verification-no-match-policy`. (Optional, `ignore`, `warn` or `fail`, default to `ignore`):
+ * `ignore`: Don't fail the taskrun/pipelinerun if no matching policies are found. Don't log.
+ * `warn`: Don't fail the taskrun/pipelinerun and log a warning if no matching policies are found.
+ * `fail`: Fail the taskrun/pipelinerun if no matching policies are found.
+
+**Note:** The current proposed `trusted-resources-verification-no-match-policy` will be added to replace the old `resource-verification-mode` in one release and this is not a backwards compatible change.
+
+### Condition Update
+
+Trusted resources should update the taskrun/pipelinerun’s condition to indicate if it passes verification or not. This can be done via knative api. Other options are listed in [alternatives](#alternatives).
+
+Need to create conditions for trusted resources so this can co-exist with current `apis.ConditionSucceeded` type:
+```go
+const (
+// ConditionTrustedResourcesVerified specifies that the resources pass trusted resources verification or not.
+ConditionTrustedResourcesVerified apis.ConditionType = "TrustedResourcesVerified"
+)
+```
+
+A successful condition is:
+ ```go
+ apis.Condition{
+			Type:    ConditionTrustedResourcesVerified,
+			Status:  corev1.ConditionTrue,
+			Reason:  podconvert.ReasonResourceVerificationSuccess
+			Message: "Trusted resource verification passed",
+		}
+ ```
+
+A failed condition is:
+  ```go
+ apis.Condition{
+			Type:    ConditionTrustedResourcesVerified,
+			Status:  corev1.ConditionFalse,
+			Reason:  podconvert.ReasonResourceVerificationFailed
+			Message: "", //filled with error message,
+		}
+ ```
+
+### How feature flag and Verification Policy update the status
+* If the `trusted-resources-verification-no-match-policy` is set to `fail`
+  * No Matching Policies found: Resources that do not have matched policies will fail the taskrun/pipelinerun.
+
+  Resulting yaml status:
+  ```yaml
+  status:
+    conditions:
+    - lastTransitionTime: "2023-03-01T18:17:05Z"
+      message: no policies are matched
+      reason: ResourceVerificationFailed
+      status: "False"
+      type: TrustedResourcesVerified
+    - lastTransitionTime: "2023-03-01T18:17:10Z"
+      message: no policies are matched
+      reason: ResourceVerificationFailed
+      status: "False"
+      type: Succeeded
+  ```
+
+   * Matching Policies found:
+     * Don't fail taskrun/pipelinerun if all the matched `enforce` policies pass or only match `warn` policies.
```

Comment:
this contradicts line 461 which says that warn policies failing causes `Conditions.TrustedResourcesVerified` to be set to "False"

Reply:
```suggestion
     * If all the matched `enforce` policies pass, and the TaskRun/PipelineRun succeeds:
```